zrml/currencies/src/lib.rs

```diff
@@ -0,0 +1,284 @@
+        }
+        fn slash_reserved(
+            currency_id: Self::CurrencyId,
+            who: &T::AccountId,
+            value: Self::Balance,
+        ) -> Self::Balance {
+            T::Currencies::slash_reserved(currency_id, who, value)
+        }
+        fn reserved_balance(currency_id: Self::CurrencyId, who: &T::AccountId) -> Self::Balance {
+            T::Currencies::reserved_balance(currency_id, who)
+        }
+        fn reserve(
+            currency_id: Self::CurrencyId,
+            who: &T::AccountId,
+            value: Self::Balance,
+        ) -> DispatchResult {
+            T::Currencies::reserve(currency_id, who, value)
+        }
+
+        fn unreserve(
+            currency_id: Self::CurrencyId,
+            who: &T::AccountId,
+            value: Self::Balance,
+        ) -> Self::Balance {
+            T::Currencies::unreserve(currency_id, who, value)
+        }
+
+        fn repatriate_reserved(
+            currency_id: Self::CurrencyId,
+            slashed: &T::AccountId,
+            beneficiary: &T::AccountId,
+            value: Self::Balance,
+            status: BalanceStatus,
+        ) -> result::Result<Self::Balance, DispatchError> {
+            T::Currencies::repatriate_reserved(currency_id, slashed, beneficiary, value, status)
+        }
+    }
+
+    impl<T: Config> MultiCurrency<T::AccountId> for Pallet<T> {
+        type CurrencyId = CurrencyIdOf<T>;
+        type Balance = BalanceOf<T>;
+        fn minimum_balance(currency_id: Self::CurrencyId) -> Self::Balance {
+            T::Currencies::minimum_balance(currency_id)
+        }
+        fn total_issuance(currency_id: Self::CurrencyId) -> Self::Balance {
+            T::Currencies::total_issuance(currency_id)
+        }
+        fn total_balance(currency_id: Self::CurrencyId, who: &T::AccountId) -> Self::Balance {
+            T::Currencies::total_balance(currency_id, who)
+        }
+        fn free_balance(currency_id: Self::CurrencyId, who: &T::AccountId) -> Self::Balance {
+            T::Currencies::free_balance(currency_id, who)
+        }
+        fn ensure_can_withdraw(
+            currency_id: Self::CurrencyId,
+            who: &T::AccountId,
+            amount: Self::Balance,
+        ) -> DispatchResult {
+            T::Currencies::ensure_can_withdraw(currency_id, who, amount)
+        }
+        fn transfer(
+            currency_id: Self::CurrencyId,
+            from: &T::AccountId,
+            to: &T::AccountId,
+            amount: Self::Balance,
+        ) -> DispatchResult {
+            Accounts::<T>::insert(currency_id, to, ());
+            T::Currencies::transfer(currency_id, from, to, amount)
+        }
+        fn deposit(
+            currency_id: Self::CurrencyId,
+            who: &T::AccountId,
+            amount: Self::Balance,
+        ) -> DispatchResult {
+            Accounts::<T>::insert(currency_id, who, ());
+            T::Currencies::deposit(currency_id, who, amount)
+        }
```

Comment:
We don't like to save the information for `ZTG`, the main asset. I think.. ?

Reply:
ZTG will not be handled by fungibles I think, it goes via fungible.

Reply:
`orml_tokens` does not handle `ZTG`, but `orml_currencies` does. https://github.com/zeitgeistpm/zeitgeist/blob/48ba93bb12cd0520c0ee0c878dc9ca0446b9454d/primitives/src/traits/zeitgeist_multi_reservable_currency.rs#L84-L88

Reply:
I see , we should add similar code in zrml currencies too.